darts/timeseries.py

```diff
@@ -649,91 +651,113 @@ def from_dataframe(
         TimeSeries
             A univariate or multivariate deterministic TimeSeries constructed from the inputs.
         """
+        df = nw.from_native(df)
 
         # get values
         if value_cols is None:
-            series_df = df.loc[:, df.columns != time_col]
+            series_df = df.drop(time_col) if time_col else df
         else:
             if isinstance(value_cols, str):
                 value_cols = [value_cols]
-            series_df = df[value_cols]
+            series_df = df[value_cols]  # quite slow
 
         # get time index
         if time_col:
             if time_col not in df.columns:
                 raise_log(AttributeError(f"time_col='{time_col}' is not present."))
 
-            time_index = pd.Index([])
             time_col_vals = df[time_col]
 
-            if np.issubdtype(time_col_vals.dtype, object):
+            if time_col_vals.dtype == nw.String:
                 # Try to convert to integers if needed
                 try:
-                    time_col_vals = time_col_vals.astype(int)
-                except ValueError:
+                    time_col_vals = time_col_vals.cast(nw.Int64)
+                except Exception:
                     pass
 
-            if np.issubdtype(time_col_vals.dtype, np.integer):
-                # We have to check all integers appear only once to have a valid index
-                raise_if(
-                    time_col_vals.duplicated().any(),
-                    "The provided integer time index column contains duplicate values.",
-                )
-
+            if time_col_vals.dtype.is_integer():
+                if time_col_vals.is_duplicated().any():
+                    raise_log(
+                        ValueError(
+                            "The provided integer time index column contains duplicate values."
+                        )
+                    )
                 # Temporarily use an integer Index to sort the values, and replace by a
                 # RangeIndex in `TimeSeries.from_xarray()`
                 time_index = pd.Index(time_col_vals)
 
-            elif np.issubdtype(time_col_vals.dtype, object):
+            elif time_col_vals.dtype == nw.String:
                 # The integer conversion failed; try datetimes
                 try:
                     time_index = pd.DatetimeIndex(time_col_vals)
```

Comment:
Asking as I don't know: how would this behave with polars/pyarrow backed series?

As long as `time_col_vals` is a narwhals series, you can try to convert it to datetime via:

```py
time_col_vals.str.to_datetime()
```

we will try to infer the date format